sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorTransmitter.cs

```diff
@@ -170,53 +180,47 @@ private MonitorBase GenerateTelemetryData(Activity activity)
                     Id = activity.Context.SpanId.ToHexString()
                 };
 
-                // TODO: Handle activity.TagObjects
-                // ExtractPropertiesFromTags(dependency.Properties, activity.Tags);
-
-                if (activityType == PartBType.Http)
+                switch (activityType)
                 {
-                    dependency.Data = UrlHelper.GetUrl(partBTags);
-                    dependency.Type = "HTTP"; // TODO: Parse for storage / SB.
-                    var statusCode = GetHttpStatusCode(partBTags);
-                    dependency.ResultCode = statusCode;
-                    dependency.Success = GetSuccessFromHttpStatusCode(statusCode);
+                    case PartBType.Http:
+                        dependency.Data = HttpHelper.GetUrl(partBTags);
+                        dependency.Target = HttpHelper.GetHost(partBTags);
+                        bool parsed = AzureBlobHttpParser.TryParse(ref dependency)
```

Comment:
This seems to be a perf hit (complexity O(N * M) where N is the number of patterns/parsers and M is the length of the input string).
I think this can be improved to O(M).